test/promql/tests/test_controller_reconciliation_errors.yaml

```
@@ -0,0 +1,133 @@
+              summary: "Controller c1-controller has reconciliation errors greater than 1%"
+              description: "Controller c1-controller with Pod pod-1 for namespace prod-ns1 has total reconciliation errors greater than 1% compared to total reconciliations in last 2 minutes."
+
+  - interval: 1m
+    input_series:
+      # Controller without error at start and then error starts. This is alerted because there is always an increase in error value
+      - series: 'controller_runtime_reconcile_errors_total{controller="c2-controller", namespace="prod-ns2", pod="pod-2", container="c2", service="s2"}'
+        values: '0 1 2'
+      - series: 'controller_runtime_reconcile_total{controller="c2-controller", namespace="prod-ns2", pod="pod-2", container="c2", service="s2"}'
+        values: '1 2 3'
+
+    alert_rule_test:
+      - eval_time: 3m
+        alertname: ControllerReconciliationErrors
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: prod-ns2
+              pod: pod-2
+              controller: c2-controller
+              container: c2
+              service: s2
+            exp_annotations:
+              summary: "Controller c2-controller has reconciliation errors greater than 1%"
+              description: "Controller c2-controller with Pod pod-2 for namespace prod-ns2 has total reconciliation errors greater than 1% compared to total reconciliations in last 2 minutes."
+
+  - interval: 1m
+    input_series:
+      # Controller with large spikes in error values and pause in reconcile total,
+      # since the error rate is always greater than 1% in 2min window this is alerted
+      - series: 'controller_runtime_reconcile_errors_total{controller="c10-controller", namespace="prod-ns10", pod="pod-10", container="c10", service="s10"}'
+        values: '1 150 390'
+      - series: 'controller_runtime_reconcile_total{controller="c10-controller", namespace="prod-ns10", pod="pod-10", container="c10", service="s10"}'
+        values: '1 500 500'
+
+    alert_rule_test:
+      - eval_time: 3m
+        alertname: ControllerReconciliationErrors
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: prod-ns10
+              pod: pod-10
+              controller: c10-controller
+              container: c10
+              service: s10
+            exp_annotations:
+              summary: "Controller c10-controller has reconciliation errors greater than 1%"
+              description: "Controller c10-controller with Pod pod-10 for namespace prod-ns10 has total reconciliation errors greater than 1% compared to total reconciliations in last 2 minutes."
+
+  # Not Alerted test cases
+  - interval: 1m
+    input_series:
+      # Controller with error throughout but reconcile error and total is always constant with values 1 and 10
+      # even though the error rate is greater than 1% it's not alerted since the error is not increasing
```

If I understand correctly, those two metrics are counters. Meaning, that if their values are constant, it means no activity. e.g. if error series is `1 1 1`, it means that there was an error some time before the first `1`, and then no errors.

I'm ok with the test case, but I think the comment should be different:
```suggestion
      # Controller with no reconciliation events throughout the test time frame.
```